Testing/MqttTests.cs

```diff
@@ -64,12 +69,11 @@ public async Task SendD2CWithRetainFlagTrue()
 
             // Act
             await device.SendDeviceToCloudMessageAsync(payload, true);
-
             var messages = await receiver.ReceiveMessagesFromDeviceAsync(new CancellationTokenSource(3000).Token);
 
             // Assert - verify message was received + mqtt-retain set to true
-            var sentMessage = messages.FirstOrDefault(x => x.Payload == payload);
-            Assert.IsTrue(sentMessage != null && sentMessage.RetainFlag == "true");
+            var sentMessage = messages[payload];
```

Comment:
If key is not present, this line will throw an error. Please add a key validation first.